pyproject.toml

```diff
@@ -1,3 +1,6 @@
+[build-system]
+requires = ["setuptools", "wheel"]
```

Comment:
what does this do?

Reply:
Making us work with https://www.python.org/dev/peps/pep-0517/ a little better (although not completely since we need to disable it for editable installs)

Reply:
For future reference: I'd really prefer stuff like this to happen in separate PRs. It's hard enough to get back up to speed when re-reviewing a big PR after 3 weeks, so anything that can be done to cut extraneous noise out is a win; likewise it means that there can be a clear record of when/why changes like this were made without it being lost in a much larger PR.

same goes for things like changing the .gitignore and dropping `DelayedCall.debug`.